src/layouts/sidebar-sidecar/utils/_local_platform-docs-mdx/components/enterprise-alert/index.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { ReactNode } from 'react'
+import classNames from 'classnames'
+
+import { useCurrentProduct } from 'contexts'
+import Badge from 'components/badge'
+import ProductIcon from 'components/product-icon'
+import Text from 'components/text'
+import InlineLink from 'components/inline-link'
+import { ProductSlug } from 'types/products'
+import { isProductSlug, productSlugsToNames } from 'lib/products'
+import s from './enterprise-alert.module.css'
+
+interface EnterpriseAlertProps {
+	productSlug: ProductSlug
+	inline?: boolean
+	className?: string
+	children?: ReactNode
+}
+
+//
+// Base components need to be slightly modified
+// to fit our use cases in dev-portal
+//
+export function EnterpriseAlert({
+	productSlug,
+	inline,
+	className,
+	children,
+}: EnterpriseAlertProps) {
+	// This ensures we aren't producing invalid HTML when rendering inline alerts within MDX. When used inline, we might end up nesting a div inside of a p. This is invalid as p cannot contain block-level elements (ref: https://www.w3.org/TR/html401/struct/text.html#h-9.3.1).
+	const Element = inline ? 'span' : 'div'
+	const isValidProduct = isProductSlug(productSlug)
+
+	if (productSlug === 'hcp') {
+		return null
+	}
+
+	if (!isValidProduct) {
+		throw new Error(
+			`[EnterpriseAlert]: Invalid product option passed. Expected one of ${Object.keys(
+				productSlugsToNames
+			).filter((slug) => slug !== 'hcp')}`
+		)
+	}
+
+	return (
+		<Element
+			className={classNames(s.root, className, {
+				[s.inline]: inline,
+			})}
+		>
+			<Badge
+				className={s.badge}
+				icon={<ProductIcon productSlug={productSlug} />}
+				text="Enterprise"
+				type="outlined"
+				size="medium"
+			/>
+			{!inline && (
+				<Text size={200} className={s.text}>
+					{children ? (
+						children
+					) : (
+						<>
+							This feature requires{' '}
+							<InlineLink
+								href={`https://www.hashicorp.com/products/${productSlug}`}
+								target="_blank"
+								rel="noopener noreferrer"
+								className={s.link}
+								textSize={200}
+							>
+								{productSlugsToNames[productSlug]} Enterprise
+							</InlineLink>
+							.
+						</>
+					)}
+				</Text>
+			)}
+		</Element>
+	)
+}
+
+export function DocsEnterpriseAlert(
+	props: Omit<EnterpriseAlertProps, 'productSlug'> & {
+		product?: ProductSlug
+	}
```

Comment:
Could be a better way to do this. I wanted only the product prop to be optional. The `Partial` utility makes them all optional but that isn't necessarily what I want here.

Reply:
🙋🏻 Was renaming the property from `productSlug` to `product` intentional?

💭 I think we could simplify to get the same output like this, it works like an overwrite so `Omit` isn't needed:

```suggestion
	props: EnterpriseAlertProps & {
		productSlug?: ProductSlug
	}
```

---

🎨 Optional -- I see value in pulling this `&` logic out and naming it (such as `DocsEnterpriseAlertProps`), so there's less cognitive load required to figure out what's tied to the `props` variable:

```tsx
type DocsEnterpriseAlertProps = EnterpriseAlertProps & {
	productSlug?: ProductSlug
}

export function DocsEnterpriseAlert(props: DocsEnterpriseAlertProps) {
	const currentProduct = useCurrentProduct()
	return (
		<EnterpriseAlert
			productSlug={props.productSlug || currentProduct.slug}
			{...props}
		/>
	)
}
```

Reply:
Yeah the reason I kept `product` here is to keep it backwards compatible with the [original implementation](https://react-components.vercel.app/components/enterprisealert) from react-components so that none of the content breaks.

👌🏻 I can separate out the interface!